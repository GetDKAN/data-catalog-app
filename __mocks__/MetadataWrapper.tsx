import { MetadataContext } from "@civicactions/data-catalog-components";

export function MetadataWrapper({testValue, children}) {
  return(
    <MetadataContext.Provider value={testValue}>
      {children}
    </MetadataContext.Provider>
  )
}

export const simpleMetadata = {
  identifier: "1234-abcd",
  issued: "2025-03-22",
  modified: "2025-03-30",
  publisher: {
    data: {
      name: "DKAN Publisher"
    }
  },
  license: "https://opendatacommons.org/licenses/by/1.0/",
  contactPoint: {
    fn: "John Doe",
    hasEmail: "mailto:data@example.org"
  }
};
