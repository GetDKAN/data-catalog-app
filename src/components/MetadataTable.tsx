import { useContext } from "react";
import { MetadataContext } from "@civicactions/data-catalog-components";
import UnknownLink from "./UnknownLink";

const MetadataTable = () => {
  const metadataContext = useContext(MetadataContext);
  const metadata = metadataContext.metadata;
  if (!metadata) {
    return null;
  }

  const hasEmail = metadata.contactPoint?.hasEmail;

  const thClasses: string = "border-b border-gray-200 p-4 pt-0 pb-3 pl-8 text-left font-medium text-gray-400 dark:border-gray-600 dark:text-gray-200";
  const tdClasses: string = "border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400";
  return (
    <div>
      <h2 className="font-bold text-lg mb-4">Additional Information</h2>
      <table className="table-auto">
        <thead>
          <tr>
            <th className={thClasses}>Field</th>
            <th className={thClasses}>Value</th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td className={tdClasses}>Publisher</td>
            <td className={tdClasses}>{metadata.publisher?.data.name}</td>
          </tr>
          <tr>
            <td className={tdClasses}>Identifier</td>
            <td className={tdClasses}>{metadata.identifier}</td>
          </tr>
          <tr>
            <td className={tdClasses}>Issued</td>
            <td className={tdClasses}>{metadata.issued}</td>
          </tr>
          <tr>
            <td className={tdClasses}>Last Update</td>
            <td className={tdClasses}>{metadata.modified}</td>
          </tr>
          {metadata.license
            && (
              <tr>
                <td className={tdClasses}>License</td>
                <td className={tdClasses}>
                  <UnknownLink url={metadata.license} text={metadata.license} />
                </td>
              </tr>
            )
          }
          <tr>
            <td className={tdClasses}>Contact</td>
            <td className={tdClasses}>{metadata.contactPoint?.fn}</td>
          </tr>
          {hasEmail
            && (
              <tr>
                <td className={tdClasses}>Contact Email</td>
                <td className={tdClasses}>
                  {hasEmail.startsWith("mailto:")
                    ? <a href={hasEmail}>{hasEmail.replace("mailto:", "")}</a>
                    : <a href={`mailto:${hasEmail}`}>{hasEmail}</a>
                  }
                </td>
              </tr>
            )
          }
          
          <tr>
            <td className={tdClasses}>Public Access Level</td>
            <td className={tdClasses}>{metadata.accessLevel}</td>
          </tr>
        </tbody>
      </table>
    </div>
  )
}

export default MetadataTable;