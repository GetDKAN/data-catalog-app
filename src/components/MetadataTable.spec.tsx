import "@vitest/browser/matchers.d.ts";
import { vi, test, expect } from "vitest";
import { MetadataWrapper, simpleMetadata } from "../../__mocks__/MetadataWrapper";
import MetadataTable from "./MetadataTable";
import { render } from 'vitest-browser-react';

vi.mock('MetadataWrapper');

test("renders", async() => {
  const { getByRole, getByText } = render(
    <MetadataWrapper testValue={{metadata: simpleMetadata}}>
      <MetadataTable />
    </MetadataWrapper>
  );
  await expect.element(getByRole("heading", {name: "Additional Information", level: 2})).toBeInTheDocument();
  await expect.element(getByText("John Doe")).toBeInTheDocument();
});

test("snapshot matches", () => {
  const screen = render(
    <MetadataWrapper testValue={{metadata: simpleMetadata}}>
      <MetadataTable />
    </MetadataWrapper>
  );
  expect(screen).toMatchSnapshot();
});
