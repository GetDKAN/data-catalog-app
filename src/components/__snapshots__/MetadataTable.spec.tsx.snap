// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot matches 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    
    
    
  


    <div>
      <div>
        <h2
          class="font-bold text-lg mb-4"
        >
          Additional Information
        </h2>
        <table
          class="table-auto"
        >
          <thead>
            <tr>
              <th
                class="border-b border-gray-200 p-4 pt-0 pb-3 pl-8 text-left font-medium text-gray-400 dark:border-gray-600 dark:text-gray-200"
              >
                Field
              </th>
              <th
                class="border-b border-gray-200 p-4 pt-0 pb-3 pl-8 text-left font-medium text-gray-400 dark:border-gray-600 dark:text-gray-200"
              >
                Value
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                Publisher
              </td>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                DKAN Publisher
              </td>
            </tr>
            <tr>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                Identifier
              </td>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                1234-abcd
              </td>
            </tr>
            <tr>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                Issued
              </td>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                2025-03-22
              </td>
            </tr>
            <tr>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                Last Update
              </td>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                2025-03-30
              </td>
            </tr>
            <tr>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                License
              </td>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                <a
                  href="https://opendatacommons.org/licenses/by/1.0/"
                  rel="noopener"
                  target="_blank"
                >
                  https://opendatacommons.org/licenses/by/1.0/
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-arrow-up-right-from-square ml-2"
                    data-icon="arrow-up-right-from-square"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 512 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l82.7 0L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3l0 82.7c0 17.7 14.3 32 32 32s32-14.3 32-32l0-160c0-17.7-14.3-32-32-32L320 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"
                      fill="currentColor"
                    />
                  </svg>
                </a>
              </td>
            </tr>
            <tr>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                Contact
              </td>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                John Doe
              </td>
            </tr>
            <tr>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                Contact Email
              </td>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                <a
                  href="mailto:data@example.org"
                >
                  data@example.org
                </a>
              </td>
            </tr>
            <tr>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              >
                Public Access Level
              </td>
              <td
                class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
              />
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="font-bold text-lg mb-4"
      >
        Additional Information
      </h2>
      <table
        class="table-auto"
      >
        <thead>
          <tr>
            <th
              class="border-b border-gray-200 p-4 pt-0 pb-3 pl-8 text-left font-medium text-gray-400 dark:border-gray-600 dark:text-gray-200"
            >
              Field
            </th>
            <th
              class="border-b border-gray-200 p-4 pt-0 pb-3 pl-8 text-left font-medium text-gray-400 dark:border-gray-600 dark:text-gray-200"
            >
              Value
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              Publisher
            </td>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              DKAN Publisher
            </td>
          </tr>
          <tr>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              Identifier
            </td>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              1234-abcd
            </td>
          </tr>
          <tr>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              Issued
            </td>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              2025-03-22
            </td>
          </tr>
          <tr>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              Last Update
            </td>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              2025-03-30
            </td>
          </tr>
          <tr>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              License
            </td>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              <a
                href="https://opendatacommons.org/licenses/by/1.0/"
                rel="noopener"
                target="_blank"
              >
                https://opendatacommons.org/licenses/by/1.0/
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-arrow-up-right-from-square ml-2"
                  data-icon="arrow-up-right-from-square"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l82.7 0L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3l0 82.7c0 17.7 14.3 32 32 32s32-14.3 32-32l0-160c0-17.7-14.3-32-32-32L320 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"
                    fill="currentColor"
                  />
                </svg>
              </a>
            </td>
          </tr>
          <tr>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              Contact
            </td>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              John Doe
            </td>
          </tr>
          <tr>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              Contact Email
            </td>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              <a
                href="mailto:data@example.org"
              >
                data@example.org
              </a>
            </td>
          </tr>
          <tr>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            >
              Public Access Level
            </td>
            <td
              class="border-b border-gray-100 p-4 pl-8 text-gray-500 dark:border-gray-700 dark:text-gray-400"
            />
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholder": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
